server/src/main/java/org/elasticsearch/cluster/metadata/ComponentTemplate.java

```diff
@@ -35,25 +36,28 @@ public class ComponentTemplate implements SimpleDiffable<ComponentTemplate>, ToX
     private static final ParseField TEMPLATE = new ParseField("template");
     private static final ParseField VERSION = new ParseField("version");
     private static final ParseField METADATA = new ParseField("_meta");
+    private static final ParseField DEPRECATED = new ParseField("deprecated");
 
     @SuppressWarnings("unchecked")
     public static final ConstructingObjectParser<ComponentTemplate, Void> PARSER = new ConstructingObjectParser<>(
         "component_template",
         false,
-        a -> new ComponentTemplate((Template) a[0], (Long) a[1], (Map<String, Object>) a[2])
+        a -> new ComponentTemplate((Template) a[0], (Long) a[1], (Map<String, Object>) a[2], (Boolean) a[3])
     );
 
     static {
         PARSER.declareObject(ConstructingObjectParser.constructorArg(), Template.PARSER, TEMPLATE);
         PARSER.declareLong(ConstructingObjectParser.optionalConstructorArg(), VERSION);
         PARSER.declareObject(ConstructingObjectParser.optionalConstructorArg(), (p, c) -> p.map(), METADATA);
+        PARSER.declareBoolean(ConstructingObjectParser.optionalConstructorArg(), DEPRECATED);
     }
 
     private final Template template;
     @Nullable
     private final Long version;
     @Nullable
     private final Map<String, Object> metadata;
+    private final boolean deprecated;
```

Comment:
One question I have is whether we want to distinguish between set and unset values. Here it means that any time we get the template back out, even if not set, it'll have a `deprecated: false` output. I generally don't like adding things unless the user also added them, but I don't have super strong feelings about it yet.

Reply:
The way I implemented `toXContent`, I'm not adding a `deprecated` flag unless it is `true`. So if a user explicitly adds `deprecated: false`, it will come out without a deprecated flag.

I get the point about preserving what the user has submitted. I hadn't really thought  about that.

My reasoning was to avoid having to deal with null values as much as possible and to make it obvious that null means false.

A compromise would be to store a `Boolean` in the class but return a `boolean` from the getter so that callers don't need to guess what `null` means and don't have guard agains potential NPEs.

Reply:
> A compromise would be to store a Boolean in the class but return a boolean from the getter so that callers don't need to guess what null means and don't have guard agains potential NPEs.

That's what I've implemented in e93ab49